### ISSUE TITLE
Support for charts in Asciidoc plug-in.

### ISSUE THREAD

**Issue by wfouche** [asciidoctor/asciidoctor-intellij-plugin#367]
This is more a question, than a bug report. I'd like to create charts in Asciidoc in IntelliJ. With the Asciidoc extension in Chrome the following chart syntax is supported:

````[chart,line]
....
January,February,March
28,48,40
65,59,80
....
````

![image](https://user-images.githubusercontent.com/13682850/68837113-8e5c6d00-06c4-11ea-98de-7846a653666e.png)

How can charts be created in Asciidoc (IntelliJ)?

**Comment by wfouche** [comment]
An initiative to support charts in Asciidoctor is discussed here:

- https://discuss.asciidoctor.org/How-to-include-bar-chart-td6030.html

- https://github.com/asciidoctor/asciidoctor-extensions-lab/pull/76

**Comment by ahus1** [comment]
Is your example missing a block macro name `[chart]` just before the "...."? 

You can load an Asciidoctor ruby extension like we describe here: https://github.com/asciidoctor/asciidoctor-intellij-plugin/wiki/Support-for-Asciidoctor-Extensions

Does this work for you?

**Comment by wfouche** [comment]
Yes, my example was missing the chart specification line. It should have read:

````
[chart,line]
....
January,February,March
28,48,40
65,59,80
....
````

I'll try the proposed workaround, and will report back. It would nice though to have chart functionality supported out-of-the-box.

**Comment by ahus1** [comment]
@Mogztter - I see that the Asciidoctor browser extension includes [some bundled extensions](https://github.com/asciidoctor/asciidoctor-browser-extension/tree/master/app/js/vendor). What were your criteria to include an extension? AsciidoctorFX as an editor has also lots of extensions. 

I fear that users see that something works in one editor and not in another. Once they use the command line none of them works as the would need to add he extension on the command line.

I'd love to hear your thoughts. Thanks.

**Comment by Mogztter** [comment]
> I see that the Asciidoctor browser extension includes some bundled extensions. 

The main issue is that a browser extension is a restricted environment. As far as I know it's not possible to register arbitrary extensions. That's why I chose to bundle a few extensions.

> What were your criteria to include an extension? AsciidoctorFX as an editor has also lots of extensions.

To be honest, I don't really have strict criteria. I chose to bundle "popular" extensions that can be useful to the majority of the users (but in the end that's quite arbitrary).
In my opinion, charts and diagrams are popular and everyone ❤️ emoji right? 

> I fear that users see that something works in one editor and not in another. Once they use the command line none of them works as the would need to add he extension on the command line.

True.
And since the syntax is consistent with other "built-in" features, it's understandable to assume that this syntax will work everywhere.

>  It would nice though to have chart functionality supported out-of-the-box

Thought decision!
I think that the support for Asciidoctor extensions in IntelliJ is great because if `.asciidoctor/lib` is versioned, it will work for everyone working on this repository.

I think the chart extension should be moved to a dedicated repository with a Ruby and a JavaScript implementation.

**Comment by ahus1** [comment]
There is already a chart block in the extension lab. If some wants to add it as a PR, I'll merge it. I'm happy to advise anyone who wants to take on this talk.
